node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -664,6 +700,8 @@ impl MainReactor {
                         };
                         Ok(Some(SyncBackInstruction::Sync {
                             parent_hash: *parent_hash,
+                            state_root_hash: *highest_orphaned_block_header.state_root_hash(),
+                            protocol_version: highest_orphaned_block_header.protocol_version(),
```

Comment:
Hmm, I think this part might break. If `highest_orphaned_block_header` is the switch block of era 1, `era_id` will be 0 here, which will cause us to land in the branch that tries to read the validators from the global state. However, the global state of the switch block in era 1 won't have the validators for era 1, so we will land in line 329 above :grimacing:
This won't be an issue unless the switch block is the only block in era 1... It shouldn't happen in most networks, but it technically could happen.

Reply:
Hmm, yes, I think you're right. But that would require that an emergency upgrade was done starting with era 1, right? And for the era to have a single switch block it wouldn't it mean it upgraded to 1.5?

Reply:
Hmm, seems like we're back at "_The alternate solution with extended sync leap may turn out to be the most robust._"